src/vm/eventpipeprovider.cpp

```diff
@@ -152,6 +153,11 @@ EventPipeProviderCallbackData EventPipeProvider::UnsetConfiguration(
 
     if (m_sessions & sessionMask)
         m_sessions &= ~sessionMask;
+
+    m_keywords = keywordsForAllSessions;
+    m_providerLevel = providerLevelForAllSessions;
+
+    RefreshAllEvents(m_keywords, m_providerLevel);
```

Comment:
To ensure no confusion I commented on #25232 that this change partially mitigates that issue but does not fully solve it. The title and description of this PR are entirely accurate, I'm just heading off the potential for it to be misconstrued.